crates/rethnet_eth/src/remote/methods.rs

```diff
@@ -0,0 +1,616 @@
+    /// eth_getTransactionReceipt
+    #[serde(
+        rename = "eth_getTransactionReceipt",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    GetTransactionReceipt(B256),
+    /// eth_mining
+    #[serde(rename = "eth_mining")]
+    Mining(),
+    /// eth_newBlockFilter
+    #[serde(rename = "eth_newBlockFilter")]
+    NewBlockFilter(),
+    /// eth_newFilter
+    #[serde(
+        rename = "eth_newFilter",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    NewFilter(FilterOptions),
+    /// eth_newPendingTransactionFilter
+    #[serde(rename = "eth_newPendingTransactionFilter")]
+    NewPendingTransactionFilter(),
+    /// eth_pendingTransactions
+    #[serde(rename = "eth_pendingTransactions")]
+    PendingTransactions(),
+    /// eth_sendRawTransaction
+    #[serde(
+        rename = "eth_sendRawTransaction",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SendRawTransaction(ZeroXPrefixedBytes),
+    /// eth_sendTransaction
+    #[serde(
+        rename = "eth_sendTransaction",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SendTransaction(TransactionInput),
+    /// eth_sign
+    #[serde(rename = "eth_sign")]
+    Sign(Address, ZeroXPrefixedBytes),
+    /// eth_signTypedData_v4
+    #[serde(rename = "eth_signTypedData_v4")]
+    SignTypedDataV4(Address, eip712::Message),
+    /// eth_subscribe
+    #[serde(
+        rename = "eth_subscribe",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    Subscribe(Vec<String>),
+    /// eth_syncing
+    #[serde(rename = "eth_syncing")]
+    Syncing(),
+    /// eth_uninstallFilter
+    #[serde(
+        rename = "eth_uninstallFilter",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    UninstallFilter(U256),
+    /// eth_unsubscribe
+    #[serde(
+        rename = "eth_unsubscribe",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    Unsubscribe(Vec<ZeroXPrefixedBytes>),
+}
+
+/// for specifying the inputs to eth_getLogs
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetLogsInput {
+    /// starting block for get_logs request
+    pub from_block: BlockSpec,
+    /// ending block for get_logs request
+    pub to_block: BlockSpec,
+    /// address for get_logs request
+    pub address: Address,
+}
+
+#[cfg(test)]
+mod tests {
+    use bytes::Bytes;
+
+    use super::*;
+
+    fn help_test_method_invocation_serde(call: MethodInvocation) {
+        let json = serde_json::json!(call).to_string();
+
+        // validate that variations of MethodInvocation which have single values can still be
```

Comment:
Is this testing Rethnet behaviour or serde behaviour?

Reply:
It's verifying that we're using `sequence_to_single`/`single_to_sequence` on all of the `MethodInvocation` variants that only have a single parameter. see more context [here](https://github.com/NomicFoundation/hardhat/pull/3967#discussion_r1218522112).

Reply:
That's already validated by request -> json -> request though, right?

Reply:
No because if the variant failed to specify both of those helpers, then the request->json part would yield `params` as a single value and not an array, and then json->request part would deserialize from the single (non-array) `params` value back to the single value of the variant.